packages/components/src/Switch.js

```diff
@@ -6,11 +6,12 @@ const GUTTER = 2
 const SIZE = 18
 
 export const Switch = React.forwardRef(function Switch(
-  { className, label, sx, variant = 'switch', ...props },
+  { className, label, labelPosition = 'end', sx, variant = 'switch', ...props },
   ref
 ) {
-  return (
-    <Label sx={{ cursor: 'pointer' }}>
+  // Inner really checkbox (but hidden).
+  const ReallyHiddenCheckbox = ({ ref, label, ...props }) => {
```

Comment:
Creating new components within components breaks reconciliation. Could you move this to the module scope or even inline it again?

In this particular case, there won't be any visible bug, but it's still a good idea to just avoid nesting components like this.

I've made a small repro for you to illustrate the problem: https://codesandbox.io/s/lucid-rain-oeq4ph?file=/src/App.tsx

Notice that when you change the text in the buggy input it resets — it mounts again, because from React's perspective this is a new component.


Further reading: https://reactjs.org/docs/reconciliation.html

Reply:
Done!